pkg/api/v1/types.go

```diff
@@ -3810,6 +3810,10 @@ type ConfigMap struct {
 	// Each key must be a valid DNS_SUBDOMAIN with an optional leading dot.
 	// +optional
 	Data map[string]string `json:"data,omitempty" protobuf:"bytes,2,rep,name=data"`
+
+	// Data contains the binary data.
```

Comment:
Needs a much more descriptive comment, including a description of the validation rules (must not overlap with data) and how lookups are done from a config map.  I'd also recommend adding additional description to the config map volume source that describes how this works.